bench/tiered/push_pull.c

```diff
@@ -0,0 +1,271 @@
+{
+    char home_full[HOME_BUF_SIZE];
+
+    printf("Running %s test \n", suffix);
+
+    testutil_check(__wt_snprintf(home_full, HOME_BUF_SIZE, "%s_%s_%d", opts->home, suffix, flush));
+    run_test(home_full, num_records, flush);
+
+    /* Cleanup */
+    if (!opts->preserve)
+        testutil_clean_work_dir(home_full);
+}
+
+/*
+ * run_test --
+ *     TODO: Add a comment describing this function.
+ */
+static void
+run_test(const char *home, uint64_t num_records, bool flush)
+{
+    struct timeval start, end;
+
+    char buf[1024];
+    double diff_sec;
+    int64_t file_size;
+
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+
+    testutil_make_work_dir(home);
+    if (opts->tiered_storage) {
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/bucket", home));
+        testutil_make_work_dir(buf);
+    }
+
+    testutil_wiredtiger_open(opts, home, conn_config, NULL, &conn, false);
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Create and populate table. Checkpoint the data after that. */
+    testutil_check(session->create(session, opts->uri, table_config_row));
+
+    testutil_check(__wt_snprintf(buf, sizeof(buf), flush ? "flush_tier=(enabled,force=true)" : ""));
+
+    gettimeofday(&start, 0);
+    populate(session, num_records);
+
+    testutil_check(session->checkpoint(session, buf));
+    gettimeofday(&end, 0);
+
+    diff_sec = difftime_sec(start, end);
+
+    sleep(1);
+
+    get_file_size(home, &file_size);
+    testutil_assert(diff_sec > 0);
+    printf("Bytes Written - %" PRIi64 ", Time took - %f seconds, Throughput - %f MB/second\n",
+      file_size, diff_sec, ((file_size / diff_sec) / WT_MEGABYTE));
+}
+
+/*
+ * populate --
+ *     Populate the table.
+ */
+static void
+populate(WT_SESSION *session, uint64_t num_records)
+{
+    WT_CURSOR *cursor;
+    WT_RAND_STATE rnd;
+    uint64_t i, str_len;
+
+    __wt_random_init_seed((WT_SESSION_IMPL *)session, &rnd);
+
+    str_len = sizeof(data_str) / sizeof(data_str[0]);
+    for (i = 0; i < str_len - 1; i++)
+        data_str[i] = 'a' + (uint32_t)__wt_random(&rnd) % 26;
+
+    data_str[str_len - 1] = '\0';
+
+    testutil_check(session->open_cursor(session, opts->uri, NULL, NULL, &cursor));
+    for (i = 0; i < num_records; i++) {
+        cursor->set_key(cursor, i + 1);
+        cursor->set_value(cursor, data_str);
+        testutil_check(cursor->insert(cursor));
+    }
+
+    testutil_check(cursor->close(cursor));
+    cursor = NULL;
+}
+
+/*
+ * compute_tiered_file_size --
+ *     Iterate over all the tiered files and compute file size..
+ */
+static void
+compute_tiered_file_size(const char *home, const char *tablename, int64_t *file_size)
+{
+    char stat_path[512];
+    int index;
+    struct stat stats;
+
+    for (index = 2; index < MAX_TIERED_FILES; ++index) {
+        testutil_check(__wt_snprintf(
+          stat_path, sizeof(stat_path), "%s/%s-%10.10d.wtobj", home, tablename, index));
```

Comment:
Why does the index start at 2?  Is the ....000001.wtobj file missing, or is it not being counted for some other reason?

Oh I see, it's being counted elsewhere.  I think this is confusing, it's probably better to count from 1 here and not count the 000001 file elsewhere.  This function only needs to be called when `opts->tiered_storage` is true.

Kind of related to this, you should initialize `*file_size` to 0 in this function.

Reply:
Fixed, now computing the file size of wt files and tiered files separately.